### PR TITLE
transformers example 

### LINKED ISSUE #960: Port "bert multi lingual tpu training (8 cores)" to Ignite
## 🚀 Feature

Recently included example of TPU usage with Ignite includes a training on a single TPU.
Idea is to port this kaggle kernel: https://www.kaggle.com/abhishek/bert-multi-lingual-tpu-training-8-cores to Ignite and include it in Ignite's show-case

Based on https://github.com/pytorch/ignite/issues/952#issuecomment-617609519

### PR DESCRIPTION
**Pull request by ahmedo42**
Fixes #960

Description:
Transformers example using huggingface and Ignite , works on TPU & GPU , tried to stick as much as possible to the code in 
[CIFAR10 example](https://github.com/pytorch/ignite/tree/master/examples/contrib/cifar10) .
Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

File: examples/contrib/transformers/README.MD [hunk 1]
```diff
@@ -0,0 +1,168 @@
+# Transformers Example with Ignite
```
Comment (vfdev-5) [inline examples/contrib/transformers/README.MD:1]: Maybe, README.md and not .MD ?

File: examples/contrib/transformers/README.MD [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+- on 1 or more GPUs or TPUs
+- compute training/validation metrics
+- log learning rate, metrics etc
+- save the best model weights
+
+Configurations:
+
+- [x] single GPU
+- [x] multi GPUs on a single node
+- [x] multi GPUs on multiple nodes
```
Comment (vfdev-5) [inline examples/contrib/transformers/README.MD:14]: Let's remove "multi GPUs on multiple nodes" case as a less relevant and tested

File: examples/contrib/transformers/README.MD [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+!curl -q https://raw.githubusercontent.com/pytorch/xla/master/contrib/scripts/env-setup.py -o pytorch-xla-env-setup.py
+!python pytorch-xla-env-setup.py --version $VERSION > /dev/null
+```
+
+```python
+from main import run
+run(backend="xla-tpu", nproc_per_node=8)
+```
+
+#### Multiple nodes, multiple GPUs
```
Comment (vfdev-5) [inline examples/contrib/transformers/README.MD:98]: Let's remove this section.

File: examples/contrib/transformers/README.MD [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+# using torch.distributed.launch
+python -u -m torch.distributed.launch --nproc_per_node=2 --use_env main.py run --backend="nccl" --stop_iteration=1000
+```
+
+Resume from the latest checkpoint
+
+```bash
+python -u -m torch.distributed.launch --nproc_per_node=2 --use_env main.py run --backend="nccl" \
+    --resume-from=/tmp/output-imdb/resnet18_backend-nccl-2_stop-on-1000/training_checkpoint_1000.pt
+```
```
Comment (vfdev-5) [inline examples/contrib/transformers/README.MD:156]: Let's remove this section as well as is not helpful for this example, I think

File: examples/contrib/transformers/dataset.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import torch
+from torch._C import dtype
```
Comment (vfdev-5) [inline examples/contrib/transformers/dataset.py:2]: Importing private member is not a good practice, I think you can use `torch.dtype` instead ?

File: examples/contrib/transformers/main.py [hunk 6]
```diff
@@ -0,0 +1,403 @@
+import os
+from datetime import datetime
+from pathlib import Path
+
+import fire
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import utils
+from torch._C import dtype
```
Comment (vfdev-5) [inline examples/contrib/transformers/main.py:10]: Same here

File: examples/contrib/transformers/main.py [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+    manual_seed(config["seed"] + rank)
+    device = idist.device()
+
+    logger = setup_logger(name="IMDB-Training", distributed_rank=local_rank)
+
+    log_basic_info(logger, config)
+
+    output_path = config["output_dir"]
+    if rank == 0:
+        if config["stop_iteration"] is None:
```
Comment (vfdev-5) [inline examples/contrib/transformers/main.py:37]: Let's remove everything related to `stop_iteration`.

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @ahmedo42 ! Looks good !
I left few comment to improve docs and the code.
Let me try to run it locally and see how it works...

**Comment by vfdev-5** [comment]
@ahmedo42 I updated a bit more the example (maybe same things should be done on cifar10 as well).
I wonder if you checked the example from your side ? Batch size being 128 is rather large, which GPU do you have :)
Let me know what do you think about these changes. Thanks

**Comment by ahmedo42** [comment]
> @ahmedo42 I updated a bit more the example (maybe same things should be done on cifar10 as well).

this definitely is cleaner , I agree that same changes should be done to the CIFAR10 example 
> I wonder if you checked the example from your side ? Batch size being 128 is rather large, which GPU do you have :)

yes, I checked on GPU , TPU on Kaggle and Colab ( as I Don't have GPU 😄 ) , the default batch size (128) was meant for TPU .

**Comment by vfdev-5** [comment]
Could you please rerun the check from your side to ensure that the example is working. I checked on GPU(s). Maybe, what remains is TPUs...

**Review by vfdev-5 — APPROVED** [review]
LGTM! Thanks @ahmedo42 !

**Comment by ahmedo42** [comment]
It works on TPU

**Pull request merged**